05-onchain.md

```diff
@@ -305,16 +305,19 @@ There are several possible cases for an offered HTLC:
    HTLC. In this case, the recipient must use the preimage, once it receives it
    from the outgoing HTLC; otherwise, it will lose funds by sending an outgoing
    payment without redeeming the incoming payment.
+4. The HTLC is forwarded as above, but the recipient also happens to be the
+   final destination for the payment.  In this case, it knows the preimage but
```

Comment:
This feels awkward - maybe "final destination for the payment, and it knows the preimage bug it has not received the full payment. In this case, it must not reveal it"?